test/com/google/enterprise/adaptor/fs/WindowsFileDelegateTest.java

```diff
@@ -936,6 +937,113 @@ public boolean pushRecord(DocIdPusher.Record record) {
     delegate.destroy();
   }
 
+  @Test
+  public void testMonitorBackOffOnError() throws Exception {
```

Comment:
Can this unit test be run against the existing code, and does it fail? Does it throw an exception, or is it just missing the change to file3 at the end because the monitor thread threw an exception that killed the thread?